doc/gdb-debugging.md

```diff
@@ -0,0 +1,85 @@
+### GDB Debugging
+This document describes the options available for debugging node using the GNU debugger.
+
+We don't provide the debuginfo package, which contains the symbols required for debugging, as this would make our runtime image about 480MB larger. On systems where root access is allowed then the debuinfo package can simply be installed using yum/rpm, but on others like OpenShift getting a pod to run with root access can be somewhat tricky. 
+
+### OpenShift
+When running an application on OpenShift the pod will not have root access. It is possible to configure such access but that means configuring the OpenShift environment. This section provides an alternative to that which allows debugging the application using a remote gdb session. This is done by starting a container in the same cluster with an image with the node runtime and debuginfo, and then connecting to the pod that is the target of the debugging.
+
+The example used in this document is [nodejs-rest-http](https://github.com/bucharest-gold/nodejs-rest-http).
+
+#### Start/attach to the running process
+First step is to start the gdbserver on the pod that is the target of the debugging session:
+```
+$ oc rsh nodejs-rest-http-1-nw7hz
+```
+We want to attach the `gdbserver` to the node process (`30` in this case but can differ):
+```console
+sh-4.2$ ps -ef
+UID        PID  PPID  C STIME TTY          TIME CMD
+1000120+     1     0  0 03:58 ?        00:00:00 npm
+1000120+    30     1  0 03:58 ?        00:00:00 node .
+1000120+    41     0  0 03:59 ?        00:00:00 /bin/sh
+1000120+    45    41  0 03:59 ?        00:00:00 ps -ef
+$ gdbserver 172.17.0.6:7777 --attach 30
```

Comment:
Is this the IP address of the pod?

Reply:
Yep

Reply:
Is there a better way of getting the ip adress of the pod by the way?

Reply:
Since we are already on the container how about we just add a note about using `hostname -i`? I've updated with a commit suggesting this.

Reply:
yea,  that probably works better